midenvm-verifier/src/main.rs

```diff
@@ -0,0 +1,63 @@
+use miden_verifier::{verify, ProgramInfo, Kernel};
+use std::env;
+use std::path::PathBuf;
+use std::path::Path;
+// use miden::verify;
+use crate::helpers::ProgramHash;
+use crate::helpers::InputFile;
+use crate::helpers::OutputFile;
+use crate::helpers::ProofFile;
+mod helpers;
+
+
+fn main() {
+    let args: Vec<String> = env::args().collect();
+    if args.len() != 5 {
+        eprintln!("Usage: {} <program_hash> <proof_path> <stack_inputs> <stack_outputs>", args[0]);
+        std::process::exit(1);
+    }
+
+    // read program hash from input
+    let program_hash = ProgramHash::read(&args[1]).unwrap();
+
+
+    // load input data from file
+    let mut input_path = PathBuf::new();
+
+    input_path.push(args[3].clone());
+
+    let proof_path = Path::new(&args[2]);
+
+    let input_data = InputFile::read(&Some(input_path), proof_path).unwrap();
+    
+    // fetch the stack inputs from the arguments
+    let stack_inputs = input_data.parse_stack_inputs().unwrap();
+    
+    // load outputs data from file
+
+    let mut output_path = PathBuf::new();
+
+    output_path.push(args[4].clone());
+
+    let outputs_data = OutputFile::read(&Some(output_path), proof_path).unwrap();
+    
+    let stack_outputs = outputs_data.stack_outputs().unwrap();
+
+    // load proof from file
+    let proof = ProofFile::read(&Some(proof_path.to_path_buf()), proof_path).unwrap();
+
+    // This is copied from core midenvm verifier.
+    // TODO accept kernel as CLI argument -- this is not done in core midenVM
```

Comment:
Kernel is used for defining syscalls. Default kernel is empty (there are no syscalls available for miden programs with default kernel). I guess we want to use it from the cli since somebody could define they own kernel. I'd need more time to investigate on this and test miden with custom kernels in order to make this work.

I'm not sure exactly how it works exactly, but some docs are here: https://crates.io/crates/miden-assembly under "Program kernels" and here: https://0xpolygonmiden.github.io/miden-vm/user_docs/assembly/execution_contexts.html?highlight=kernel#kernels

From my research it seems like it is for special purpose code - maybe some optimized circuits for hash functions etc.

Note that miden core also uses default kernel in their cli (for now...?).